source/extensions/filters/common/fault/fault_config.h

```diff
@@ -44,28 +49,42 @@ class FaultAbortConfig {
 
     // Return the HTTP status code to use. Optionally passed an HTTP header that may contain the
     // HTTP status code depending on the provider implementation.
-    virtual absl::optional<Http::Code> statusCode(const Http::HeaderEntry* header) const PURE;
+    virtual absl::optional<Http::Code> httpStatusCode(const Http::HeaderEntry* header) const PURE;
+
+    // Return the gRPC status code to use. Optionally passed an HTTP header that may contain the
+    // gRPC status code depending on the provider implementation.
+    virtual absl::optional<Grpc::Status::GrpcStatus>
+    grpcStatusCode(const Http::HeaderEntry* header) const PURE;
   };
 
   // Delay provider that uses a fixed abort status code.
   class FixedAbortProvider : public AbortProvider {
   public:
-    FixedAbortProvider(uint64_t status_code) : status_code_(status_code) {}
+    FixedAbortProvider(absl::optional<Http::Code> http_status_code,
```

Comment:
i think it would be better if http_status_code was not optional here. Otherwise I think you need error checking for when both parameters are null.